source/features/avoid-accidental-submissions.tsx

```diff
@@ -9,32 +10,48 @@ function addQuickSubmit(): void {
 }
 
 function onKeyDown(event: delegate.Event<KeyboardEvent, HTMLInputElement>): void {
+	const form = event.delegateTarget.form!;
 	if (
 		event.key !== 'Enter'
 		|| event.ctrlKey
 		|| event.metaKey
 		|| event.isComposing // #4323
-		|| select.exists('.suggester', event.delegateTarget.form!) // GitHub’s autocomplete dropdown
+		|| select.exists([
+			'.suggester', // GitHub’s autocomplete dropdown
+			'.rgh-avoid-accidental-submissions.my-1',
+		], form)
 	) {
 		return;
 	}
 
+	if (!select.exists('.btn-primary[type="submit"]:not(:disabled)', form)) {
+		return;
+	}
+
+	const focusedInput = select(inputElements, form)!;
```

Comment:
I think you just need to use `event.delegateTarget`, no need to reselect the item. The changes around `select().focus()` need to be undone

Reply:
I have followed your suggestions in the new commit.
Tested it in all cases and it works.